CE 23.2.1 Breaks Wayland support on Linux

### Description

The latest version of DBeaver seems to have inadvertently regressed. Suddenly, X11 has become a hard dependency, although X11 is itself deprecated technology and it will be faced out over time.

```
Caused by: java.awt.HeadlessException: 
No X11 DISPLAY variable was set,
but this program performed an operation which requires it.
	at java.desktop/java.awt.Desktop.getDesktop(Desktop.java:302)
	at org.jkiss.awt.injector.ProxyInjector.injectBrowseInteraction(ProxyInjector.java:31)
	at org.jkiss.dbeaver.core.DBeaverActivator.injectProxyPeer(DBeaverActivator.java:80)
	at org.jkiss.dbeaver.core.DBeaverActivator.start(DBeaverActivator.java:74)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:818)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:810)
	... 30 more
```

[CE_23.2.1_CrashLogWayland.txt](https://github.com/dbeaver/dbeaver/files/12712082/CE_23.2.1_CrashLogWayland.txt)

The log has two start-ups: First with the default Wayland-first configuration, and then with wayland disabled.

### DBeaver Version

Community Edition 23.2.1

### Operating System

Fedora Linux 38, Flatpak

### Database and driver

Not applicable

### Steps to reproduce

Take the latest test-build from Flathub:
- https://github.com/flathub/io.dbeaver.DBeaverCommunity/pull/187
- Start on a Wayland-enabled platform
- Witness the crash
- (Optional) disable Wayland support and enable X11, see it start

### Additional context

_No response_

--- a/plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/DBeaverActivator.java
+++ b/plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/DBeaverActivator.java
@@ -20,6 +20,7 @@ import org.eclipse.core.runtime.Platform;
 import org.eclipse.jface.resource.ImageDescriptor;
 import org.eclipse.ui.plugin.AbstractUIPlugin;
 import org.jkiss.awt.injector.ProxyInjector;
+import org.jkiss.dbeaver.DBeaverPreferences;
 import org.jkiss.dbeaver.ModelPreferences;
 import org.jkiss.dbeaver.model.impl.preferences.BundlePreferenceStore;
 import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
@@ -28,6 +29,7 @@ import org.jkiss.dbeaver.ui.browser.BrowsePeerMethods;
 import org.osgi.framework.Bundle;
 import org.osgi.framework.BundleContext;
 
+import java.awt.*;
 import java.io.PrintStream;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
@@ -69,11 +71,17 @@ public class DBeaverActivator extends AbstractUIPlugin {
         } catch (MissingResourceException x) {
             coreResourceBundle = null;
         }
-        if (DesktopPlatform.isStandalone()) {
+        if (DesktopPlatform.isStandalone() && getPreferenceStore().getBoolean(DBeaverPreferences.UI_USE_EMBEDDED_AUTH)) {
             try {
-                injectProxyPeer();
+                if (Desktop.isDesktopSupported()) {
+                    injectProxyPeer();
+                } else {
+                    getLog().warn("Desktop interface not available");
+                    getPreferenceStore().setValue(DBeaverPreferences.UI_USE_EMBEDDED_AUTH, false);
+                }
             } catch (Throwable e) {
-                e.printStackTrace();
+                getLog().warn(e.getMessage());
+                getPreferenceStore().setValue(DBeaverPreferences.UI_USE_EMBEDDED_AUTH, false);
             }
         }
     }

